// @mkproject/common/build/events/base-listiner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Listiner = void 0;
class Listiner {
    constructor(client) {
        this.ackWait = 5 * 1000;
        this.client = client;
    }
    subscriptionOption() {
        return this.client
            .subscriptionOptions()
            .setDeliverAllAvailable()
            .setManualAckMode(true)
            .setAckWait(this.ackWait)
            .setDurableName(this.queueGroupName);
    }
    listen() {
        const subescription = this.client.subscribe(this.subject, this.queueGroupName, this.subscriptionOption());
        subescription.on("message", (msg) => {
            console.log(`message recieved: ${this.subject}  / ${this.queueGroupName}`);
            const parsedData = this.parseMessage(msg);
            this.onMessage(parsedData, msg);
        });
    }
    parseMessage(msg) {
        const data = msg.getData();
        return typeof data === "string" ? JSON.parse(data) : JSON.parse(data.toString("utf-8"));
    }
}
exports.Listiner = Listiner;
